### PR TITLE
Issue #10971: Add example to show how to configure the check with different format for final and non-final pattern variables

### LINKED ISSUE #10971: LocalFinalVariableName: false negative on pattern variable
https://checkstyle.org/config_naming.html#LocalFinalVariableName

````
$ cat TestClass.java
public class Test {
    public void run(String... arguments) {
        final Object o = 45;
        if (o instanceof final String BAD) {
            System.out.println(BAD);
        }
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="LocalFinalVariableName"/>
<module name="LocalVariableName"/>
    </module>
</module>

$ java -jar checkstyle-9.0.1-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

I am expecting a violation on line 5 since `BAD` is a local, final variable and it doesn't start with a lower case as is default required.

```
public class Test {
    public void run(String... arguments) {
        final Object o = 45;
        if (o instanceof final String BAD) {
            o = 34;  // javac: Test.java:5: error: cannot assign a value to final variable o
            System.out.println(BAD);
        }
    }
}
```

### PR DESCRIPTION
**Pull request by mahfouz72**
closes #10971

### CODE REVIEW

**Comment by mahfouz72** [comment]
Github, generate site

**Comment by mahfouz72** [comment]
@rnveach ping

**Review by rnveach — APPROVED** [review]

**Review by rnveach — CHANGES_REQUESTED** [review]
From issue ( https://github.com/checkstyle/checkstyle/issues/10971#issuecomment-2147920381 ) :

> Also your example only showed how to suppress one form of patterns. To validate both, you would have to add 2 instances of PatternVariableName and 2 SuppressionXpathSingleFilter, both with different IDs, to ensure everything is validated.

> previously mentioned documentation update that the local variable checks do not cover patterns, and to go to the pattern version of the check

These don't seem to be done.

**Comment by mahfouz72** [comment]
@rnveach done
- updated example 4 to show how to validate both (final and non final) with different format
- updated doc for `LocalVariableNameCheck` to point to `PatternVariableNameCheck`
- updated doc for `FinalLocalVariableNameCheck` to point to `PatternVariableNameCheck`

**Comment by mahfouz72** [comment]
github, generate site

File: src/xdocs/checks/naming/localvariablename.xml.template [hunk 1]
```diff
@@ -13,6 +13,11 @@
           Checks that local, non-<code>final</code> variable names conform to a specified pattern.
           A catch parameter is considered to be a local variable.
         </p>
+        <p>
+          This check does not support pattern variables. Instead, use
+          <a href="https://checkstyle.org/checks/naming/patternvariablename.html#PatternVariableName">
```
Comment (rnveach) [inline src/xdocs/checks/naming/localvariablename.xml.template:18]: links in our documentation must be relative if they are for our own domain.

File: src/xdocs/checks/naming/localfinalvariablename.xml.template [hunk 2]
```diff
@@ -14,6 +14,11 @@
           A catch parameter and resources in try statements
           are considered to be a local, final variables.
         </p>
+        <p>
+          This check does not support final pattern variables. Instead, use
+          <a href="https://checkstyle.org/checks/naming/patternvariablename.html#PatternVariableName">
```
Comment (rnveach) [inline src/xdocs/checks/naming/localfinalvariablename.xml.template:19]: same.

**Review by rnveach — CHANGES_REQUESTED** [review]

**Comment by mahfouz72** [comment]
github, generate site

File: src/xdocs/checks/naming/localvariablename.xml.template (hunk 1 above)
Comment (mahfouz72) [reply src/xdocs/checks/naming/localvariablename.xml.template]: The relative link is not working can you correct me ?

Comment (mahfouz72) [reply src/xdocs/checks/naming/localvariablename.xml.template]: @rnveach please help the link is not working now. Can you help me what is wrong?

**Comment by mahfouz72** [comment]
github, generate site

File: src/xdocs/checks/naming/localvariablename.xml.template (hunk 1 above)
Comment (mahfouz72) [reply src/xdocs/checks/naming/localvariablename.xml.template]: Yeah, It should work now. but still link check plugin is falling is this expected? I think it don't recognize the relative paths and always violate them right?

Comment (rnveach) [reply src/xdocs/checks/naming/localvariablename.xml.template]: > `+<a href="patternvariablename.html#PatternVariableName">patternvariablename.html#PatternVariableName</a>: doesn't exist.`
`+<a href="patternvariablename.html#PatternVariableName">patternvariablename.html#PatternVariableName</a>: doesn't exist.`

These shouldn't be showing up.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalFinalVariableNameCheck.java [hunk 3]
```diff
@@ -30,6 +30,11 @@
  *  A catch parameter and resources in try statements
  * are considered to be a local, final variables.
  * </p>
+ * <p>
+ * This check does not support final pattern variables. Instead, use
+ * <a href="patternvariablename.html#PatternVariableName">
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalFinalVariableNameCheck.java:35]: Javadoc has to be fully qualified URL. Only xdoc can't be.

**Review by rnveach — CHANGES_REQUESTED** [review]

**Comment by mahfouz72** [comment]
github, generate site

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalFinalVariableNameCheck.java (hunk 3 above)
Comment (mahfouz72) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalFinalVariableNameCheck.java]: Yeah, make sense. done

Comment (mahfouz72) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalFinalVariableNameCheck.java]: xwiki failure is unrelated

**Comment by mahfouz72** [comment]
Rebased

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalFinalVariableNameCheck.java (hunk 3 above)
Comment (mahfouz72) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalFinalVariableNameCheck.java]: @rnveach ping

File: src/xdocs/checks/naming/localvariablename.xml.template (hunk 1 above)
Comment (rnveach) [reply src/xdocs/checks/naming/localvariablename.xml.template]: @mahfouz72 ping. This is still not done.

**Review by rnveach — APPROVED** [review]

**Pull request merged**